### PR TITLE
[DC-1058] Modify Numeric PPI response cleaning rule to include "6+" generalization

### PR DESCRIPTION
**Pull request by dcampbell-vumc**
* Modified query to include new AOU concept code
* Modified integration test to properly test changes to cleaning rule

### CODE REVIEW

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
I left question for Caroline and Heather on the ticket.  The test works as designed, but fails on edge cases.  I'm attaching a patch file that updates this test and the 11+ test with edge case testing.  Apply it, view the message, and create your fix.
[clean_ppi_edge_case_tests.txt](https://github.com/all-of-us/curation/files/5211358/clean_ppi_edge_case_tests.txt)

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
Your business logic in `data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py` is broken based on the criteria in your ticket.  The test patch was provided to hightlight the broken business logic.

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters_test.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ def setUp(self):
         self.value_as_number = None
         self.value_as_concept_id = 2000000010
         self.dc_1061_value_as_concept_id = 2000000013
+        self.dc_1058_value_as_concept_id = 2000000012
```
[the remark this answers is not in the document]
Comment (dcampbell-vumc) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters_test.py:65]: You're right, I wouldn't even remember what is happening after a week :p, I changed them to have more descriptive variable names

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters_test.py [hunk 2]
```diff
@@ -135,11 +148,20 @@ def test_invalid_values_fields_cleaning(self):
...
+                               (765, 0, 6, 222, 1333015),
+                               (654, 0, self.value_as_number, 222, 1333015),
+                               (6543, 0, 10, 222, 1333015),
+                               (111, 0, self.value_as_number,
+                                self.six_plus_value_as_concept_id, 1333023),
+                               (222, 0, 5, 444, 1333023),
+                               (333, 0, self.value_as_number,
+                                self.six_plus_value_as_concept_id, 1585890),
+                               (444, 0, 5, 444, 1585890),
+                               (543, 0, self.value_as_number, 543, 1585890)],
```
Comment (lrwb-aou) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters_test.py:164]:
```suggestion
                               (543, 0, self.value_as_number, self.six_plus_value_as_concept_id, 1585890)],
```

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
The code in the cleaning rule is wrong.  I added the correct test condition.  I expect it to fail until the rule is corrected.

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
This is my fault.  I didn't have the comment on the ticket completely correct.  I have updated that comment and am providing a patch file to make the changes here.  Please, try it and see if it makes sense with what we know we are supposed to be seeing.
[clean_ppi_patch.txt](https://github.com/all-of-us/curation/files/5234709/clean_ppi_patch.txt)

File: data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py [hunk 3]
```diff
@@ -62,7 +66,12 @@
     WHEN observation_concept_id = 1585890 AND (value_as_number < 0 OR value_as_number > 20) THEN NULL
     WHEN observation_concept_id IN (1585795, 1585802, 1585864, 1585870, 1585873, 1586159, 1586162) AND (value_as_number < 0 OR value_as_number > 99) THEN NULL
     WHEN observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255) THEN NULL
-    WHEN observation_source_concept_id IN (1333015, 1585889) AND (value_as_number < 0 OR value_as_number > 11) THEN NULL
+    
+    -- from dc1058: will null invalid values for value_as_number if participant household size is greater than 11 --
+    WHEN observation_source_concept_id IN (1333015, 1585889) AND (value_as_number < 0 OR value_as_number > 10) THEN NULL
+    
+    -- from dc1061: will null invalid values for value_as_number if participant household has 6 or more members under the age of 18 --
+    WHEN observation_source_concept_id IN (1333023, 1585890) AND (value_as_number < 0 OR value_as_number > 5) THEN NULL
```
[the remark this answers is not in the document]
Comment (dcampbell-vumc) [reply data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py]: Yes, this makes sense to me.

**Review by lrwb-aou — APPROVED** [review]
This looks right.  Now every test that expects a Null value for value_as_number has a generalized value for value_as_concept_id.  Thank you for working through this!
